common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerFactory.scala

```diff
@@ -52,7 +52,23 @@ case class ContainerPoolConfig(userMemory: ByteSize, concurrentPeekFactor: Doubl
  */
 trait ContainerFactory {
 
-  /** create a new Container */
+  /**
+   * Create a new Container
+   *
+   * The created container has to satisfy following requirements:
+   * - The container's file system is based on the provided action image and has a read/write layer on top.
```

Comment:
I don't think this is necessarily a requirement. Blackbox containers wouldn't need that.

Reply:
@markusthoemmes I guess you talk about the read/write layer on top? Without that layer, not a single change can be made to existing files / directories and nothing new can be created.

I guess most managed runtime images require creation of new files. And I think that blackbox image authors also expect that they can create files when initialising / running the action.

Reply:
They might from a feature standpoint, yes. It is not required for the system to work in general. From a spec's PoV i'd say we make this a "SHOULD" or "MAY" in that: 

> The container's file system is based on the provided action image and may have a read/write layer on top.

Reply:
A read only file system is nice from a security standpoint: the container image is immutable.

Reply:
I like the read only and nonroot by default. 

I believe AWS Lambda the file system to the function is read only, and only have access to small of write access in /tmp which I have to guess is just a mount volume in the container.

Reply:
it would be a nice enhancement although the way we "init" some functions wrinkles this.

Reply:
Let's not get ourselves tangled up in enhancements and feature requests here please. For the sake of this documentation I would state:

> The container's file system is based on the provided action image and may have a read/write layer on top.

Reply:
I support @markusthoemmes' point of view. This is not a feature discussion but a description to clarify how things work today.

Does anybody know whether any of the prominent runtimes NEED a writable layer? If yes, this should be mentioned somehow.

Reply:
Node.js will need it if using zips.

Reply:
i think they all do: node/python for zip files, swift/go for binaries, java/ballerina for jar file.

Reply:
for the feature discussion there is this open issue (which I agree 💯 is orthogonal to the goals of this PR): https://github.com/apache/incubator-openwhisk/issues/890